### PR TITLE
[MOSIO-34112] Adding test cases

### PR DESCRIPTION
**Pull request by Khuddusshariff0022**

### CODE REVIEW

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 1]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+                .thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107394")
+                .thenReturn("9403107395")
+                .thenReturn("9403107396");
+
+        ReflectionTestUtils.setField(bioDedupeProcessor, "biometricsUpdateSingleMatchAutoReject", false);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: We should use the ArgumentCaptor as well

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 2]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107397")
+                .thenReturn("9403107398")
+                .thenReturn("9403107399");
+
+        ReflectionTestUtils.setField(bioDedupeProcessor, "biometricsUpdateSingleMatchAutoReject", false);
+
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: We should use the ArgumentCaptor as well

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 3]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        registrationStatusDto.setRegistrationType("UPDATE");
+        registrationStatusDto.setStatusCode("PROCESSING");
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        Mockito.when(abisHandlerUtil.getPacketStatus(any()))
+                .thenReturn(AbisConstant.POST_ABIS_IDENTIFICATION)
+                .thenReturn(AbisConstant.DUPLICATE_FOR_SAME_TRANSACTION_ID);
+
+        Set<String> matchedRidList = new HashSet<>();
+        matchedRidList.add("27847657360002520190320095011");
+        matchedRidList.add("27847657360002520190320095011");
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: Both RIDs should be different.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 4]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(utility.getUIn(anyString(), anyString(), any()))
+                .thenReturn("");
+
+        ReflectionTestUtils.setField(bioDedupeProcessor, "biometricsUpdateSingleMatchAutoReject", false);
+
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: We should use the ArgumentCaptor as well

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 5]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(utility.getUIn(anyString(), anyString(), any()))
+                .thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107397")
+                .thenReturn("")
+                .thenReturn("9403107398");
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: We should use the ArgumentCaptor as well

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 6]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(utility.getUIn(anyString(), anyString(), any()))
+                .thenReturn("9403107396");
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("")
+                .thenReturn("")
+                .thenReturn("");
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertFalse(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: We should use the ArgumentCaptor as well

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java [hunk 7]
```diff
@@ -185,9 +185,7 @@ public MessageDTO process(MessageDTO object, String stageName) {
 					newPacketPreAbisIdentification(registrationStatusDto, object);
 				} else if (packetStatus.equalsIgnoreCase(AbisConstant.POST_ABIS_IDENTIFICATION)) {
 					postAbisIdentification(registrationStatusDto, object, registrationType);
-
 				}
-
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java:190]: Remove these extra changes

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 8]
```diff
@@ -450,11 +456,11 @@ public void testBioDeDupUpdatePacketHandlerProcessingSuccess() throws ApisResour
 	 * @throws RegistrationProcessorCheckedException
 	 */
 	@Test
-	@Ignore
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java:453]: Shall we remove this test case or just keep @Ignore annotation.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 9]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+        Mockito.verify(registrationStatusService).updateRegistrationStatus(argumentCaptor.capture(),anyString(),anyString());
+        Assert.assertEquals(argumentCaptor.getValue().getStatusCode(), RegistrationStatusCode.PROCESSING.name());
+        assertTrue(messageDto.getIsValid());
+        assertFalse(messageDto.getInternalError());
+    }
+
+//    One match of RID/AID found at bio dedupe,
+//    but UIN of the duplicate match is different from update packet UIN when AutoReject is True
+    @Test
+    public void oneMatchFoundWithDiffrentUinWithMatchAutoRejecAsTrue() throws ApisResourceAccessException, IOException,
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: Correct the typo mistake Diffrent -> Different. Check other places as well.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java [hunk 10]
```diff
@@ -728,4 +734,296 @@ public void testBioDeDupUpdatePacketHandlerProcessingFailWithAutoRejectEnable()
...
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095011");
+        registrationStatusDto.setRegistrationType("UPDATE");
```
Comment (ashok-ksharma) [inline registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java:769]: Add the below for currently processing RID.
registrationStatusDto.setStatusCode("PROCESSING");
Check other occurrences as well.

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java (hunk 7 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java:190]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 8 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java:453]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 9 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: Done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 10 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java:769]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 1 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 2 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 4 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 6 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 5 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

File: registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java (hunk 3 above)
Comment (Khuddusshariff0022) [reply registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java]: done

**Comment by ashok-ksharma** [comment]
Code reviewed.

**Review by vishwa-vyom — APPROVED** [review]
approving based on Ashok's approval

**Pull request merged**